app/backend/src/couchers/servicers/notifications.py

```diff
@@ -102,25 +103,25 @@ def GetVapidPublicKey(self, request, context):
 
         return notifications_pb2.GetVapidPublicKeyRes(vapid_public_key=get_vapid_public_key())
 
-    def RegisterPushNotification(self, request, context):
+    def RegisterPushNotificationSubscription(self, request, context):
```

Comment:
how this `RegisterPushNotificationSubscription` function works exactly? it's an http sent to the frontend and the frontend it's listening? (through the service worker?)

or is this some websocket connection? (just to understand what's going on)

Reply:
This is exposing the backend grpc function so I can tell the backend that push notifications are registered once they subscribe via the buttons on the frontend.

It will insert a row into the `push_notification_subscriptions` table, I believe. Aapeli did all the backend though so I'm a little iffy. He just renamed this function and pushed it to my branch.

It's not a web socket because that would only send push notifications while the app was open. The cool thing about using a service worker is it can tell the user they got a new message even if they are not on our website at that moment, so it reminds them to go back to the website and check their message. It's kinda like we're running in circles around our app yelling things at it lol.

Reply:
oh wow, that's cooler than what I though! I was expecting notifications only when the couchers tab was open, that's cool, good to know!!